### PR TITLE
Feature/carousel

### PR DESCRIPTION
**Pull request by iruzevic**
- Adding generic carousel block. 
- Adding option on wrapper to have parent class

### CODE REVIEW

File: blocks/init/src/Blocks/custom/carousel/assets/carousel-slider.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+		this.element = options.element;
+		this.blockClass = options.blockClass;
+		this.nextElement = options.nextElement;
+		this.prevElement = options.prevElement;
+		this.paginationElement = options.paginationElement;
+		this.eventName = options.eventName;
+	}
+
+	init() {
+		this.initCarousel();
```
Comment (dingo-d) [inline blocks/init/src/Blocks/custom/carousel/assets/carousel-slider.js:15]: Why the abstraction?

File: blocks/init/src/Blocks/wrapper/wrapper.php [hunk 2]
```diff
@@ -14,14 +14,25 @@
 $wrapperUse = Components::checkAttr('wrapperUse', $attributes, $manifest);
 $wrapperUseSimple = Components::checkAttr('wrapperUseSimple', $attributes, $manifest);
 $wrapperDisable = Components::checkAttr('wrapperDisable', $attributes, $manifest);
+$wrapperParentClass = Components::checkAttr('wrapperParentClass', $attributes, $manifest);
 
 if (! $wrapperUse || $wrapperDisable) {
+	if ($wrapperParentClass) {
+		echo '<div class="' . \esc_attr($wrapperParentClass . '__item') . '">';
+			echo '<div class="' . \esc_attr($wrapperParentClass . '__item-inner') . '">';
```
Comment (dingo-d) [inline blocks/init/src/Blocks/wrapper/wrapper.php:22]:
```suggestion
		echo '<div class="' , \esc_attr($wrapperParentClass . '__item') , '">
			<div class="' , \esc_attr($wrapperParentClass . '__item-inner') , '">';
```

File: blocks/init/src/Blocks/wrapper/wrapper.php [hunk 3]
```diff
@@ -14,14 +14,25 @@
...
+
 	$this->renderWrapperView(
 		$templatePath,
 		$attributes,
 		$innerBlockContent
 	);
 
+	if ($wrapperParentClass) {
+			echo '</div>';
+		echo '</div>';
```
Comment (dingo-d) [inline blocks/init/src/Blocks/wrapper/wrapper.php:33]:
```suggestion
			echo '</div>
		</div>';
```

**Review by dingo-d — CHANGES_REQUESTED** [review]
One small `echo` related change.

File: blocks/init/src/Blocks/custom/carousel/carousel-style.scss [hunk 4]
```diff
@@ -0,0 +1,22 @@
+@import '~swiper/swiper.scss';
+
+.block-carousel {
+	$this: &;
```
Comment (IvanGrginovInf) [inline blocks/init/src/Blocks/custom/carousel/carousel-style.scss:4]: unnecessary this

**Review by IvanGrginovInf — CHANGES_REQUESTED** [review]

File: blocks/init/src/Blocks/custom/carousel/assets/carousel-slider.js (hunk 1 above)
Comment (iruzevic) [reply blocks/init/src/Blocks/custom/carousel/assets/carousel-slider.js:15]: no sure I will remove it

**Comment by iruzevic** [comment]
All changes are fixed

**Review by IvanGrginovInf — APPROVED** [review]

File: blocks/init/src/Blocks/wrapper/wrapper.php (hunk 3 above)
Comment (dingo-d) [reply blocks/init/src/Blocks/wrapper/wrapper.php:33]: ⬆️  Missed this one?

**Review by dingo-d — CHANGES_REQUESTED** [review]
Seems you missed one echo comment

**Pull request merged**